src/fortran/cantera_kinetics.f90

```diff
@@ -114,11 +114,12 @@ double precision function ctkin_productStoichCoeff(self, k, i)
       ctkin_productstoichcoeff = kin_productstoichcoeff(self%kin_id, k, i)
     end function ctkin_productstoichcoeff
 
-    integer function ctkin_reactionType(self, i)
+    integer function ctkin_reactionType(self, i, nm)
       implicit none
       type(phase_t), intent(in) :: self
       integer, intent(in) :: i
-      ctkin_reactiontype = kin_reactiontype(self%kin_id, i)
+      character*(*), intent(out) :: nm
+      ctkin_reactiontype = kin_reactiontype(self%kin_id, i, nm)
```

Comment:
I think the pattern this needs to follow is the one of `ctkin_getReactionString` (and I guess, the new function name ought to be `ctkin_getReactionType`).

It's a little unfortunate to have an abrupt API change like this, but I guess this is a rarely-used function and it is more allowable given we're jumping to a new major version.

Reply:
:+1: ... I agree that this is likely seldom used and ok with a major jump. But I *think* a deprecation warning has been triggered in 2.6, although the suggested workaround wasn't available for clib and f90.

Reply:
Good point, users would have gotten a warning from the C++ layer, so the only difficulty is not having a soft transition path.

Reply:
True. It’s an interesting situation that apparently is only infrequent as the clib and f90 interfaces are incomplete.

Reply:
On a separate note, we dropped a comparable interface from Python as it received a much more detailed reaction interface. I still think that the current change is the best short-time path.